roadmap/implementers-guide/src/types/candidate.md

```diff
@@ -150,10 +130,34 @@ struct LocalValidationData {
 	/// height. This may be equal to the current perceived relay-chain block height, in
 	/// which case the code upgrade should be applied at the end of the signaling
 	/// block.
+	///
+	/// This informs a relay-chain backing check and the parachain logic.
 	code_upgrade_allowed: Option<BlockNumber>,
+
+	/// The relay-chain block number this is in the context of. This informs the collator.
+	block_number: BlockNumber,
+}
+```
+
+## NonPersistedValidationData
```

Comment:
```suggestion
## TransientValidationData
```

I'd rather use a real word than construct "nonpersisted". Bonus: save 3-4 chars every time we need to type out the name.